src/java.base/share/classes/jdk/internal/classfile/Classfile.java

```diff
@@ -28,175 +28,204 @@
 import java.lang.constant.ClassDesc;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.LinkedHashSet;
-import java.util.List;
 import java.util.function.Consumer;
 import java.util.function.Function;
 
 import jdk.internal.classfile.attribute.ModuleAttribute;
 import jdk.internal.classfile.attribute.UnknownAttribute;
 import jdk.internal.classfile.constantpool.ClassEntry;
 import jdk.internal.classfile.constantpool.ConstantPoolBuilder;
-import jdk.internal.classfile.constantpool.PackageEntry;
 import jdk.internal.classfile.constantpool.Utf8Entry;
-import jdk.internal.classfile.impl.ClassImpl;
 import jdk.internal.classfile.impl.AbstractPoolEntry;
-import jdk.internal.classfile.impl.DirectClassBuilder;
-import jdk.internal.classfile.impl.Options;
-import jdk.internal.classfile.impl.SplitConstantPool;
-import jdk.internal.classfile.impl.UnboundAttribute;
+import jdk.internal.classfile.impl.ClassfileImpl;
+import jdk.internal.classfile.impl.TemporaryConstantPool;
 import java.lang.reflect.AccessFlag;
 import jdk.internal.classfile.attribute.CharacterRangeInfo;
 import jdk.internal.classfile.attribute.LocalVariableInfo;
 import jdk.internal.classfile.attribute.LocalVariableTypeInfo;
 import jdk.internal.classfile.instruction.ExceptionCatch;
-import java.lang.constant.PackageDesc;
 
 /**
  * Main entry points for parsing, transforming, and generating classfiles.
  */
-public class Classfile {
-    private Classfile() {
+public sealed interface Classfile
+        permits ClassfileImpl {
+
+    static Classfile of() {
+        return new ClassfileImpl();
```

Comment:
We can create a static final field in `ClassfileImpl` holding a default instance equivalent to that created by `new ClassfileImpl()` and have the `of()` factory return that instance instead.

Reply:
Global static default context will hold global shared cached class hierarchy resolver and that we want to avoid.

Reply:
As the default CHR stays static instance and it is no more cached, we can have also static holder for the default Classfile instance now.